AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLiteStorageEngineAdapterTests.swift

```diff
@@ -375,4 +375,120 @@ class SQLiteStorageEngineAdapterTests: BaseDataStoreTests {
             XCTFail(String(describing: error))
         }
     }
+
+    func testClearIfNewVersionShouldStoreInitialVersion() {
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionIsEmpty") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+        userDefaults.removeObject(forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "newVersion"
+
+        let mockFileManager = MockFileManager()
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameAndFileExists() {
+        let fileExistsInvoked = expectation(description: "Version remains the same")
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionInUserDefaultsRemainsSame") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "previousVersion"
+
+        let mockFileManager = MockFileManager()
+        mockFileManager.fileExists = true
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        fileExistsInvoked.fulfill()
+
+        wait(for: [fileExistsInvoked], timeout: 10)
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameButFileDoesNotExist() {
```

Comment:
`testClearIfNewVersionWithMissingFile`
- same set up as `testClearIfNewVersionWithVersionSameAsPrevious` except your fileExists = false
- make sure fileManager.removeItem is not called

Reply:
Sorry, I always has trouble thinking about name of test function